packages/contracts-bedrock/src/L2/FeeVault.sol

```diff
@@ -56,55 +39,87 @@ abstract contract FeeVault {
     }
 
     /// @notice Network which the recipient will receive fees on.
     function withdrawalNetwork() public view returns (Types.WithdrawalNetwork network_) {
-        network_ = WITHDRAWAL_NETWORK;
+        (,, network_) = config();
+    }
+
+    /// @notice Minimum balance before a withdrawal can be triggered.
+    ///         Use the `minWithdrawalAmount()` getter as this is deprecated
+    ///         and is subject to be removed in the future.
+    /// @custom:legacy
+    function MIN_WITHDRAWAL_AMOUNT() external view returns (uint256) {
+        return minWithdrawalAmount();
+    }
+
+    /// @notice Account that will receive the fees. Can be located on L1 or L2.
+    ///         Use the `recipient()` getter as this is deprecated
+    ///         and is subject to be removed in the future.
+    /// @custom:legacy
+    function RECIPIENT() external view returns (address) {
+        return recipient();
+    }
+
+    /// @notice Network which the recipient will receive fees on.
+    ///         Use the `withdrawalNetwork()` getter as this is deprecated
+    ///         and is subject to be removed in the future.
+    /// @custom:legacy
+    function WITHDRAWAL_NETWORK() external view returns (Types.WithdrawalNetwork) {
+        return withdrawalNetwork();
     }
 
     /// @notice Triggers a withdrawal of funds to the fee wallet on L1 or L2.
     function withdraw() external {
+        (address feeRecipient, uint256 minimumWithdrawalAmount, Types.WithdrawalNetwork feeWithdrawalNetwork) = config();
         require(
-            address(this).balance >= MIN_WITHDRAWAL_AMOUNT,
+            address(this).balance >= minimumWithdrawalAmount,
             "FeeVault: withdrawal amount must be greater than minimum withdrawal amount"
         );
 
         uint256 value = address(this).balance;
         totalProcessed += value;
 
-        emit Withdrawal(value, RECIPIENT, msg.sender);
-        emit Withdrawal(value, RECIPIENT, msg.sender, WITHDRAWAL_NETWORK);
+        emit Withdrawal(value, feeRecipient, msg.sender);
+        emit Withdrawal(value, feeRecipient, msg.sender, feeWithdrawalNetwork);
 
-        if (WITHDRAWAL_NETWORK == Types.WithdrawalNetwork.L2) {
-            bool success = SafeCall.send(RECIPIENT, value);
+        if (feeWithdrawalNetwork == Types.WithdrawalNetwork.L2) {
+            bool success = SafeCall.send(feeRecipient, value);
             require(success, "FeeVault: failed to send ETH to L2 fee recipient");
         } else {
             IL2ToL1MessagePasser(payable(Predeploys.L2_TO_L1_MESSAGE_PASSER)).initiateWithdrawal{ value: value }({
-                _target: RECIPIENT,
+                _target: feeRecipient,
                 _gasLimit: WITHDRAWAL_MIN_GAS,
                 _data: hex""
             });
         }
     }
+
+    /// @notice Internal getter function for the L1Block contract.
+    /// @return Contract of the L1Block on this domain.
+    function L1_BLOCK() internal pure returns (IL1Block) {
```

Comment:
Do we really need this function? Why not call the `IL1Block(Predeploys.L1_BLOCK_ATTRIBUTES)` directly on each of the vaults? Thats the way its done across the repo